### PR TITLE
Create PCA colormap for images with channels > 3

### PR DESCRIPTION
**Pull request by chungmin99**
Current colormap supports 3-channel (RGB) and 1-channel (depth, accumulation).

It would be useful to support colormaps for n-channel images for research visualization. (e.g., images with DINO features, ...) 

@kerrj for reference

Some screenshots of the PCA colormap, for dino features:

Screenshot 2023-05-16 at 10 21 15 AM
Screenshot 2023-05-16 at 10 21 54 AM
Screenshot 2023-05-16 at 10 22 41 AM
Screenshot 2023-05-16 at 10 22 26 AM

### CODE REVIEW

**Review by SauravMaheshkar — CHANGES_REQUESTED** [review]
Please run `black` on the changed files for the CI to pass.

File: nerfstudio/utils/colormaps.py [hunk 1]
```diff
@@ -164,3 +167,40 @@ def apply_boolean_colormap(
     colored_image[image[..., 0], :] = true_color
     colored_image[~image[..., 0], :] = false_color
     return colored_image
+
+def apply_pca_colormap(image):
```
Comment (tancik) [inline nerfstudio/utils/colormaps.py:171]: Add types

**Review by tancik — COMMENTED** [review]
This is great! Can you add a screenshot of it working to the PR description?

File: nerfstudio/utils/colormaps.py [hunk 2]
```diff
@@ -80,6 +80,9 @@ def apply_colormap(
     if image.dtype == torch.bool:
         return apply_boolean_colormap(image)
 
+    if image.shape[-1] > 3:
```
Comment (tancik) [inline nerfstudio/utils/colormaps.py:83]: Can you update the docstring to inform users that PCA is used when >3 channels (line 51)

**Review by kerrj — APPROVED** [review]
lgtm

**Pull request merged**